Error in building O2 since #422

Hi, 

I am getting some errors when compiling O2 since PR #422. 

Thanks for checking.



`/alice/sw/SOURCES/O2/dev/0/Utilities/DataCompression/test/../include/DataCompression/DataDeflater.h:47:48: warning: static_assert with no message is a C++1z extension [-Wc++1z-extensions]
  static_assert(Length <= 8 * sizeof(code_type));
                                               ^
                                               , ""
/alice/sw/SOURCES/O2/dev/0/Utilities/DataCompression/test/test_DataDeflater.cxx:99:23: error: implicit instantiation of undefined template 'std::__1::array<char, 8>'
  std::array<char, 8> data = {'d','e','a','d','b','e','e','f'};
                      ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__tuple:116:65: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TYPE_VIS_ONLY array;
                                                                ^
/alice/sw/SOURCES/O2/dev/0/Utilities/DataCompression/test/test_DataDeflater.cxx:124:23: error: implicit instantiation of undefined template 'std::__1::array<char, 8>'
  std::array<char, 8> data = {'d','e','a','d','b','e','e','f'};
                      ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__tuple:116:65: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TYPE_VIS_ONLY array;
                                                                ^
/alice/sw/SOURCES/O2/dev/0/Utilities/DataCompression/test/test_DataDeflater.cxx:169:23: error: implicit instantiation of undefined template 'std::__1::array<char, 8>'
  std::array<char, 8> data = {'d','e','a','d','b','e','e','f'};
                      ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__tuple:116:65: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TYPE_VIS_ONLY array;
                                                                ^
1 warning and 3 errors generated.`

@matthiasrichter PR #432 does not compile either even though the error is elsewhere now

Utilities/aliceHLTwrapper/src/Component.cxx:169:13: error: use of undeclared identifier 'ENOKEY'
    return -ENOKEY;

--- a/Utilities/DataCompression/include/DataCompression/DataDeflater.h
+++ b/Utilities/DataCompression/include/DataCompression/DataDeflater.h
@@ -44,7 +44,7 @@ template<typename CodeType, std::size_t Length = 8 * sizeof(CodeType)>
 class CodecIdentity {
 public:
   using code_type = CodeType;
-  static_assert(Length <= 8 * sizeof(code_type));
+  static_assert(Length <= 8 * sizeof(code_type), "CodeType must allow specified bit length");
   static const std::size_t sMaxLength = Length;
 
   CodecIdentity() = default;

--- a/Utilities/DataCompression/test/test_DataDeflater.cxx
+++ b/Utilities/DataCompression/test/test_DataDeflater.cxx
@@ -32,6 +32,7 @@
 #include <iostream>
 #include <iomanip>
 #include <vector>
+#include <array>
 #include <bitset>
 #include <thread>
 #include <stdexcept>  // exeptions, runtime_error

